### ISSUE TITLE
[Feature Request] Support `stop_gradient` globally 

### ISSUE THREAD

**Issue by tqtifnypmb** [ml-explore/mlx#1022]
Hi

Will MLX supports `stop_gradient` globally for inference only use cases?

Thanks

**Comment by Rifur13** [comment]
MLX doesn’t need a specific mechanism (like `torch.no_grad()`) to disable gradient tracking. 

Similarly to JAX, MLX operations do not automatically accumulate gradients. To compute gradients you need to explicitly call `mx.grad`.
